plugins/flytekit-modin/flytekitplugins/modin/schema.py

```diff
@@ -0,0 +1,152 @@
+
+
+class ParquetIO(object):
+    PARQUET_ENGINE = "pyarrow"
+
+    def _read(self, chunk: os.PathLike, columns: typing.List[str], **kwargs) -> pandas.DataFrame:
+        return pandas.read_parquet(chunk, columns=columns, engine=self.PARQUET_ENGINE, **kwargs)
+
+    def read(self, *files: os.PathLike, columns: typing.List[str] = None, **kwargs) -> pandas.DataFrame:
+        frames = [self._read(chunk=f, columns=columns, **kwargs) for f in files if os.path.getsize(f) > 0]
+        if len(frames) == 1:
+            return frames[0]
+        elif len(frames) > 1:
+            return pandas.concat(frames, copy=True)
+        return pandas.DataFrame()
+
+    def write(
+        self,
+        df: pandas.DataFrame,
+        to_file: os.PathLike,
+        coerce_timestamps: str = "us",
+        allow_truncated_timestamps: bool = False,
+        **kwargs,
+    ):
+        """
+        Writes data frame as a chunk to the local directory owned by the Schema object.  Will later be uploaded to s3.
+        :param df: data frame to write as parquet
+        :param to_file: Sink file to write the dataframe to
+        :param coerce_timestamps: format to store timestamp in parquet. 'us', 'ms', 's' are allowed values.
+            Note: if your timestamps will lose data due to the coercion, your write will fail!  Nanoseconds are
+            problematic in the Parquet format and will not work. See allow_truncated_timestamps.
+        :param allow_truncated_timestamps: default False. Allow truncation when coercing timestamps to a coarser
+            resolution.
+        """
+        df.to_parquet(
+            to_file,
+            coerce_timestamps=coerce_timestamps,
+            allow_truncated_timestamps=allow_truncated_timestamps,
+            **kwargs,
+        )
+
+
+class FastParquetIO(ParquetIO):
+    PARQUET_ENGINE = "fastparquet"
+
+    def _read(self, chunk: os.PathLike, columns: typing.List[str], **kwargs) -> pandas.DataFrame:
+        from fastparquet import ParquetFile as _ParquetFile
+        from fastparquet import thrift_structures as _ts
+
+        df = pandas.read_parquet(chunk, columns=columns, engine=self.PARQUET_ENGINE, index=False)
+        df_column_types = df.dtypes
+        pf = _ParquetFile(chunk)
+        schema_column_dtypes = {l.name: l.type for l in list(pf.schema.schema_elements)}
+
+        for idx in df_column_types[df_column_types == "float16"].index.tolist():
+            # A hacky way to get the string representations of the column types of a parquet schema
+            # Reference:
+            # https://github.com/dask/fastparquet/blob/f4ecc67f50e7bf98b2d0099c9589c615ea4b06aa/fastparquet/schema.py
+            if _ts.parquet_thrift.Type._VALUES_TO_NAMES[schema_column_dtypes[idx]] == "BOOLEAN":
+                df[idx] = df[idx].astype("object")
+                df[idx].replace({0: False, 1: True, pandas.np.nan: None}, inplace=True)
+        return df
+
+
+_PARQUETIO_ENGINES: typing.Dict[str, ParquetIO] = {

```

Comment:
Especially you do not need these separate parquet engines. We only have them, because there is a bug in Pyarrow Parquet. Presumably, modin implements its own parquet engine, or uses one of these